app/controllers/dogs_controller.rb

```diff
@@ -62,7 +62,14 @@ class DogsController < ApplicationController
   def index
     @title = session[:mgr_view] ? 'Dog Manager' : 'Our Dogs'
 
-    do_manager_view = signed_in? && session[:mgr_view]
+    if signed_in? && session[:mgr_view]
```

Comment:
could get slick and do

`do_manager_view = signed_in? && (session[:mgr_view] || params[:all_dogs] == "true")`